server/api/db/sqldb/db.py

```diff
@@ -2478,33 +2516,7 @@ def _delete_multi_objects(
 
             # The select is mandatory for sqlalchemy 1.4 because
             # query.delete does not support multiple-table criteria within DELETE
-            if project != "*":
-                subquery = (
-                    select(cls.id)
-                    .join(main_table)
-                    .where(
-                        and_(
-                            main_table.project == project,
-                            or_(
-                                main_table_identifier == value
-                                for value in main_table_identifier_values
-                            ),
-                        )
-                    )
-                    .subquery()
-                )
-            else:
-                subquery = (
-                    select(cls.id)
-                    .join(main_table)
-                    .where(
-                        or_(
-                            main_table_identifier == value
-                            for value in main_table_identifier_values
-                        )
-                    )
-                    .subquery()
-                )
+            subquery = select(cls.id).join(main_table).where(where_clause).subquery()
             stmt = (
                 delete(cls)
                 .where(cls.id.in_(aliased(subquery)))
```

Comment:
I don't fully understand why do you still need to delete the related tables manually, if we add teh CASCADE option.
I see from the docs that all of the related tables [need to be loaded](https://docs.sqlalchemy.org/en/14/orm/cascades.html#using-foreign-key-on-delete-cascade-with-orm-relationships:~:text=If%20the%20my_parent.children%20collection%20is%20unloaded%2C%20then%20no%20DELETE%20statements%20are%20emitted.%20If%20the%20relationship.passive_deletes%20flag%20were%20not%20set%20on%20this%20relationship()%2C%20then%20a%20SELECT%20statement%20for%20unloaded%20Child%20objects%20would%20have%20been%20emitted.), so it might makes sense to run the select as above, but when the parent table record is deleted, the children should be deleted automatically...

Perhaps there's something I'm missing?

Reply:
I left this to allow using this method with other relationships that are not cascading although I don't see a need for it at this time

Reply:
So won't it make the optimization redundent?
As you are still deleting the related tables first, then the main table, so you don't really optimize

Reply:
no because I don't pass the related tables in the params so they are skipped